vyper/compiler/README.md

```diff
@@ -0,0 +1,95 @@
+# 🐍 `vyper.compiler` 🐍
+
+## Purpose
+
+The `vyper.compiler` module contains the main user-facing functionality used to compile
+vyper source code and generate various compiler outputs.
+
+## Organization
+
+`vyper.compiler` has the following structure:
+
+* [`__init__.py`](__init__.py): Contains the `compile_codes` function, which is the
+primary function used for compiling Vyper source code.
+* [`data.py`](data.py): Pure functions for executing each compiler phase, as well as
+the `CompilerData` object that fetches and stores compiler output for each phase.
+* [`output.py`](output.py): Functions that convert compiler data into the final
+formats to be outputted to the user.
+* [`utils.py`](utils.py): Various utility functions related to compilation.
+
+## Control Flow
+
+### Compiler Phases
+
+The compilation process has five broad phases:
+
+1. In [`vyper.ast`](`../ast`), the source code is parsed and converted to an
+abstract syntax tree.
+2. The [`GlobalContext`](../parser/global_context.py) object is generated from the
+Vyper AST, analyzing and organizing the nodes prior to LLL generation.
+3. In [`vyper.parser.parser`](../parser/parser.py), the contextualized nodes are
```

Comment:
This name will change when we finish ripping apart parser, right?

Reply:
Definitely.  It's already such a large diff, I didn't wanna start also changing things around outside.